src/writers.ts

```diff
@@ -8,15 +14,28 @@ export function createEvmWriters(indexerName: string) {
 
     const sender = event.args.sender;
     const token = event.args.token;
-    const amount = event.args.amount;
+    const amount = (Number(event.args.amount.toString()) / 1e6).toFixed(2); // Transforms 1990000 to 1.99
     const barcode = event.args.barcode;
 
-    const payment = new Payment(`${sender}/${tx.hash}`, indexerName);
+    const chain = indexerName; // The indexer name corresponds to the chain name
+    let tokenName = getTokenName(token, chain);
+
+    if (!tokenName) {
+      // TODO log missing token
+      tokenName = token;
+    }
+
+    const payment = new Payment(`${tx.hash}`, indexerName);
     payment.sender = sender;
-    payment.token = token;
+    payment.token = tokenName;
     payment.amount = amount;
+
     payment.barcode = barcode;
+    const ipfsData = await getJSON(barcode);
```

Comment:
This doesn't seem to be used, also the naming is confusing, this is not necessary an IPFS URL / data.

Reply:
I mean right now the whole indexing relies on the fact that there's a valid JSON here, on IPFS. So... it's kind of necessarily an IPFS URL / data no ? It's from the SX-API code.

Reply:
It's not enforced, someone can use a normal URL like https://example.com/salut.json or others data storage system like swarm://0x123

Reply:
a942a437e1ebc4584e581ce5f29451d1679fbe51